frontend/views/containers/group-settings/InvitationsTable.vue

```diff
@@ -189,39 +233,41 @@ export default {
         invitee: isAnyoneLink ? L('Anyone') : invitee,
         inviteSecret,
         inviteLink: buildInvitationUrl(this.currentGroupId, inviteSecret),
+        description: this.inviteStatusDescription({
+          isAnyoneLink, isInviteExpired, isInviteRevoked, isAllInviteUsed, quantity, numberOfResponses
+        }),
+        expiryInfo: isInviteExpired ? L('Expired') : isInviteRevoked ? L('Revoked') : isAllInviteUsed ? '' : this.readableExpiryInfo(expiryTime),
         status: {
           isExpired: isInviteExpired,
-          isActive: !isInviteExpired && !isAllInviteUsed,
-          description: this.inviteStatusDescription({
-            isAnyoneLink, isInviteExpired, isAllInviteUsed, quantity, numberOfResponses
-          }),
-          expiryInfo: isAllInviteUsed ? '' : this.readableExpiryInfo(expiryTime)
+          isActive: !isInviteExpired && !isInviteRevoked && !isAllInviteUsed,
+          isRevoked: isInviteRevoked
         }
       }
     },
     handleInviteClick (e) {
       if (e.target.classList.contains('js-btnInvite')) {
         sbp('okTurtles.events/emit', OPEN_MODAL, 'AddMembers')
       }
-    }
-  },
-  computed: {
-    ...mapGetters(['currentGroupState', 'ourUsername']),
-    ...mapState(['currentGroupId']),
-    invitesToShow () {
-      const { invites } = this.currentGroupState
-
-      if (!invites) { return [] }
-
-      const invitesList = Object.values(invites)
-        .filter(invite => invite.creator === INVITE_INITIAL_CREATOR || invite.creator === this.ourUsername)
-        .map(this.mapInvite)
-      const options = {
-        Active: () => invitesList.filter(invite => invite.status.isActive),
-        All: () => invitesList
+    },
+    async handleRevokeClick (inviteSecret) {
+      if (!confirm(L('Are you sure you want to revoke this link? This action cannot be undone.'))) {
+        return null
       }
 
-      return options[this.ephemeral.selectbox.selectedOption]()
+      try {
+        this.ephemeral.inviteRevokedNow = inviteSecret
+        const message = await sbp('gi.contracts/group/inviteRevoke/create', {
+          inviteSecret
+        }, this.currentGroupId)
+        await sbp('backend/publishLogEntry', message)
+        setTimeout(() => {
```

Comment:
There's a very unlikely but real race condition here. If an exception is thrown, which sets `inviteRevokedNow` to null, and the user takes an action that sets it to non-null within these 2 seconds, then the timer created on this line is going to override the user's action by setting it to null. To solve it we'd just have to store this timer in a variable, and cancel it in the catch block.

Reply:
The only thing that can throw an exception though is on lines 259 and 262, right? So if either of those lines threw an exception we wouldn't get to line 263, and if we did get to this line it implies no exception was thrown...

Reply:
Oh good point. Oops.